php-fpm/Dockerfile

```diff
@@ -1,57 +1,155 @@
-ARG PHP_SOURCE_IMAGE="docker.io/davidalger/php"
+ARG ENV_SOURCE_IMAGE
 ARG PHP_VERSION
-ARG PHP_VARIANT="fpm-loaders"
-FROM ${PHP_SOURCE_IMAGE}:${PHP_VERSION}-${PHP_VARIANT}
+ARG OS_RELEASE
 
-# Clear undesired settings from base fpm images
-ENV COMPOSER_ALLOW_SUPERUSER=
-ENV COMPOSER_HOME=
+FROM golang:alpine AS mhs-builder
+RUN go install github.com/mailhog/mhsendmail@latest
+
+
+FROM ${ENV_SOURCE_IMAGE:-php}:${PHP_VERSION}-fpm-${OS_RELEASE:-bullseye}
+USER root
+ARG PHP_VERSION
 
 ENV MAILHOG_HOST    mailhog
 ENV MAILHOG_PORT    1025
+ENV USER_ID         501
+ENV GROUP_ID        20
+ENV OSTYPE          darwin220
 
-RUN dnf install -y which pv sudo bind-utils python3-pip bash-completion rsync socat oathtool \
-        wget ack make gcc gcc-c++ automake autoconf crontabs mariadb nano vim coreutils-common \
-    && dnf clean all \
-    && rm -rf /var/cache/dnf
+# PHP Extension Installer
+ADD https://github.com/mlocati/docker-php-extension-installer/releases/latest/download/install-php-extensions /usr/local/bin/
 
-RUN dnf install -y python2 \
-    && ln -s /bin/python2 /bin/python \
-    && dnf clean all \
-    && rm -rf /var/cache/dnf
+# Copy mhsendmail to PHP
+COPY --from=mhs-builder /go/bin/mhsendmail /usr/local/bin/mhsendmail
 
-ENV NODE_VERSION    10
-RUN npm install -g n \
-    && n install ${NODE_VERSION} \
-    && rm -rf /usr/local/n/versions/node
+# Install PHP Extensions required by Magento OS, Adobe Commerce, and the UCT (pcntl)
+RUN chmod +x /usr/local/bin/install-php-extensions && install-php-extensions \
+    apcu \
+    amqp \
+    bcmath \
+    calendar \
+    exif \
+    gd \
+    intl \
+    imagick \
+    imap \
+    mcrypt \
+    mysqli \
+    pcntl \
+    pdo_mysql \
+    redis \
+    soap \
+    sockets \
+    sodium \
+    xsl \
+    zip
+
+# Install helpful utilities
+RUN set -eux; \
+    apt-get update -y; \
+    apt-get install -y --no-install-recommends \
+    jq \
+    bash \
+    patch \
+    pv \
+    procps \
+    nano \
+    vim \
+    mariadb-client \
+    sudo \
+    gettext \
+    ca-certificates \
+    git \
+    rsync \
+    bash-completion \
+    nmap  \
+    less \
+    zsh \
+    fish \
+    pwgen \
+    openssh-client \
+    unzip \
+    rsync \
+    socat \
+    oathtool \
+    wget \
+    ack  \
+    automake  \
+    autoconf \
+    python \
+    cron \
+    htop \
+    chromium \
+    imagemagick \
+    graphicsmagick
```

Comment:
This will leave a bunch of temporary files in the layer bloating the image. You should run `apt-get clean` as well as `rm -rf /var/lib/apt/lists/*` to remove temporary files

Reply:
@bap14 added the clean and remove the temporarily apt files.